internal/agent/agent.go

```diff
@@ -146,21 +146,14 @@ func (a *Agent) Run(ctx context.Context) error {
 		a.log,
 	)
 
-	// create config controller
-	configController := config.NewController(
-		hookManager,
-		deviceReadWriter,
-		a.log,
-	)
-
 	bootstrap := device.NewBootstrap(
 		deviceName,
 		executer,
 		deviceReadWriter,
 		csr,
 		specManager,
 		statusManager,
-		configController,
+		hookManager,
```

Comment:
note: shared functionality for hooks should be exposed by the hookManager not by its consumer.